R/modules.R

```diff
@@ -103,6 +103,42 @@ append_module <- function(modules, module) {
   modules
 }
 
+#' Extract/Remove module(s) of specific class
+#'
+#' Given a `teal_module` or a `teal_modules`, return the elements of the structure according to `class`.
+#'
+#' @param modules `teal_modules`
+#' @param class The class name of `teal_module` to be extracted or dropped.
+#' @keywords internal
+#' @return For `extract_module`, a `teal_module` of class `class` or `teal_modules` containing modules of class `class`.
+#' For `drop_module`, an opposite, which is all `teal_modules` besides the ones with `class` class.
+#' @rdname extract_module
+extract_module <- function(modules, class) {
+  if (inherits(modules, class)) {
+    modules
+  } else if (inherits(modules, "teal_module")) {
+    NULL
+  } else if (inherits(modules, "teal_modules")) {
+    Filter(function(x) length(x) > 0L, lapply(modules$children, extract_module, class))
+  }
+}
+
+#' @keywords internal
+#' @return `teal_modules`
+#' @rdname extract_module
```

Comment:
Please use a different name for the file, otherwise the header will display `extract_module` regardless of which function is queried. Perhaps `manage_modules`? `module_management`? I wonder if `append_module` should be on the same page.

Shall we also add `@seealso` in the page for `modules` to direct to this topic? I'm not convinced because `modules` is exported, so it's for `teal` users, and these functions are not, so they're for `teal` developers.

Reply:
I also considered having `append_module` in the same page, but then you need to come up with one title and it really didn't play well together.

@seealso is not recommended for unexported functions.

For the consistency I can rename extract_module to `module_management` but I think this is minor. Have seen a lot of functions in R ecosystem on one page, that was named by first function. Anyway. chaning to `module_management` :)

Reply:
132872b5bd7bcba6a4deb115d6c9e5ef57101fcf

Reply:
> Have seen a lot of functions in R ecosystem on one page, that was named by first function.

In the messy parts, no doubt :wink: